back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/parsers/pdf/idea_plain_text_parser_service.rb

```diff
@@ -2,336 +2,137 @@
-        end
-
-        next if @current_custom_field.nil?
-        next if part_of_description? line
-        next if disclaimer? line
-
-        field_type = @current_custom_field.input_type
-
-        if TEXT_FIELD_TYPES.include? field_type
-          current_text = @form[:fields][@current_field_display_title]
-          @form[:fields][@current_field_display_title] = current_text.nil? ? line : "#{current_text} #{line}"
-          next
-        end
-
-        if MULTILINE_FIELD_TYPES.include? field_type
-          current_text = @form[:fields][@current_field_display_title]
-          @form[:fields][@current_field_display_title] = current_text.nil? ? line : "#{current_text} #{line}"
-          next
-        end
-
-        if field_type == 'select'
-          handle_select_field(line)
-        end
-
-        if field_type == 'multiselect'
-          handle_multiselect_field(line)
-        end
-
-        if NUMBER_FIELD_TYPES.include? field_type
-          handle_number_field(line)
+      # Extract each field by splitting the page by each field title
+      optional_copy = I18n.with_locale(@locale) { I18n.t('form_builder.pdf_export.optional') }
+      page_hash = {}
+      page_fragment = page
+      @custom_fields.reverse_each do |field|
+        title = field.title_multiloc[@locale]
+        title = "#{title} (#{optional_copy})" if field.required == false
+        text_split = page_fragment.split(title)
+        if text_split.length > 1
+          field_value = process_field_value(field, text_split[1])
+          page_hash[title] = field_value
+          page_fragment = text_split[0]
         end
       end
+      page_hash.reverse_each.to_h
     end
 
-    def get_form_title(phase)
-      phase_title = phase.title_multiloc[@locale]
-      project_title = phase.project.title_multiloc[@locale]
-
-      "#{project_title} - #{phase_title}"
-    end
+    def process_field_value(field, value)
+      # Strip out unneeded generic text
+      field_value = value.strip
 
-    def new_form
-      {
-        pdf_pages: [],
-        form_pages: [],
-        fields: {}
-      }
-    end
+      # Remove field descriptions
+      description = ActionView::Base.full_sanitizer.sanitize(field.description_multiloc[@locale])
+      field_value = field_value&.delete_prefix(description)&.strip if description.present?
 
-    def find_line_with_page_number(lines)
-      # Check last three lines for page number
-      last_index = lines.length
+      # Remove page numbers
+      page_copy = I18n.with_locale(@locale) { I18n.t('form_builder.pdf_export.page') }
+      page_number_regex = Regexp.new "#{page_copy} \\d+$"
+      field_value = field_value&.gsub(page_number_regex, '')&.strip
 
-      (1..3).each do |n|
-        line = lines[last_index - n]
-        return line if page_number? line
+      # Process different field types
+      field_type = field.input_type
+      if field_type == 'select'
+        field_value = handle_select_field(field, field_value)
```

Comment:
A `case` would look really nice here, but of course it's just a matter of taste :slightly_smiling_face: 
```suggestion
      field_value = 
        case field_type
        when 'select' then handle_select_field(field, field_value)
        when 'multiselect' then handle_multiselect_field(field, field_value)
        when *NUMBER_FIELD_TYPES then handle_number_field(field, field_value)
        end
```